tests/js/panelsTests.js

```diff
@@ -399,6 +399,112 @@ https://github.com/gpii/universal/LICENSE.txt
+        gradeNames: ["fluid.test.testCaseHolder", "autoInit"],
+        modules: [{
+            name: "Tests the keyboard panel",
+            tests: [{
+                expect: 4,
+                name: "Initialization",
+                sequence: [{
+                    func: "{keyboard}.refreshView"
+                }, {
+                    listener: "gpii.tests.keyboardTester.verifyInitRendering",
+                    args: ["{keyboard}", "keyboardInstructions"],
+                    event: "{keyboard}.events.afterRender"
+                }]
+            }, {
+                expect: 4,
+                name: "Don't Offer Assistance",
+                sequence: [{
+                    func: "{keyboard}.applier.change",
+                    args: ["offerAssistance", false]
+                }, {
+                    listener: "gpii.tests.keyboardTester.verifyNoAssistance",
+                    args: ["{keyboard}"],
+                    spec: {path: "offerAssistance", priority: "last"},
+                    changeEvent: "{keyboard}.applier.modelChanged"
+                }]
+            }, {
+                expect: 7,
+                name: "Offer Assistance",
+                sequence: [{
+                    func: "{keyboard}.applier.change",
+                    args: ["offerAssistance", true]
+                }, {
+                    listener: "gpii.tests.keyboardTester.verifyOfferAssistance",
+                    args: ["{keyboard}"],
+                    spec: {path: "offerAssistance", priority: "last"},
+                    changeEvent: "{keyboard}.applier.modelChanged"
+                }]
+            }]
+        }]
+    });
+
+    gpii.tests.keyboardTester.verifyInitRendering = function (that, instructions) {
+        jqUnit.assertEquals("The instructions should be rendered correctly", that.options.messageBase[instructions], that.locate("instructions").text());
+        jqUnit.notVisible("The assistance element should be hidden", that.locate("assistance"));
+
+        jqUnit.exists("The input should be present", that.locate("input"));
+        jqUnit.assertEquals("The placeholder text should be set correctly", that.options.messageBase.placeholder, that.locate("input").attr("placeholder"));
+
+    };
+
+    gpii.tests.keyboardTester.verifyNoAssistance = function (that) {
+        jqUnit.assertFalse("The offerAssistance model value should be false", that.model.offerAssistance);
+        jqUnit.assertEquals("The instructions should be rendered correctly", that.options.messageBase.successInstructions, that.locate("instructions").text());
+
+        jqUnit.notVisible("The assistnace should be hidden", that.locate("assistance"));
+        jqUnit.notExists("The input should be removed", that.locate("input"));
+    };
+
+    gpii.tests.keyboardTester.verifyOfferAssistance = function (that) {
+        jqUnit.assertTrue("The offerAssistance model value should be true", that.model.offerAssistance);
```

Comment:
Can you also test the states of sub-components to make sure they are created and destroyed when they need to?